### DISCUSSION TITLE
onModelBeforeRead

### DISCUSSION THREAD

**Discussion by octavonu** [pocketbase/pocketbase#2388] [Q&A]
Hello,

how would you implement a hook for DB Read (onModelBeforeRead)?

Thank you

**Comment by ganigeorgiev** [comment]
Hm, this may not be very efficient, especially when retrieving hundreds of records.

What would be the use case for that? Could you give an example?

**Reply by octavonu** [reply]
Thank you for your fast Re. and congrats for your awesome work. 

Let's suppose that an app is taking care of maintenance of equipments distributed in two or more factories.

Each field user responsible with maintenance of this app would be assigned to a specific factory. A field user can repair/order spare parts only for equipments associated to its factory.

So, in authRecord there is a value for factory and subsequently, I would like to use this in all future selects but of course without writing for each one a where factory = authRecord.(*models.Record).Get("factory")

Hope that I was clear.
Thank you

**Reply by ganigeorgiev** [reply]
Sorry, I'm still not sure that I understand the use case for this.

If you want to modify the record data after load from the database it may be better to just create your own wrapper func that accept the record(s) and update the fields from there.

Firing an app level hook for every db model load could be very inefficient and I'm not sure if it is worth the tradeoff since I don't think too many people will use it.

I've added it to my internal todo to reconsider it in the future but for now remains out of the scope for v1.
